x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/ProfilingIndexManager.java

```diff
@@ -354,7 +354,7 @@ public ProfilingIndex withGeneration(String generation) {
         }
 
         public boolean isMatchWithoutVersion(String indexName) {
-            return indexName.startsWith("." + namePrefix);
+            return indexName.startsWith("." + namePrefix + "-v");
```

Comment:
This still matches e.g. `.profiling-stackframes-very`.
You could tighten the check further by testing for the following exact three ASCII digits after the `v`.